core/src/ast_builder/select/join.rs

```diff
@@ -0,0 +1,543 @@
+use {
+    super::{NodeData, Prebuild},
+    crate::{
+        ast::{
+            Join, JoinConstraint, JoinExecutor, JoinOperator, ObjectName, Statement, TableAlias,
+            TableFactor,
+        },
+        ast_builder::{
+            ExprList, ExprNode, FilterNode, GroupByNode, JoinConstraintNode, LimitNode, OffsetNode,
+            ProjectNode, SelectItemList, SelectNode,
+        },
+        result::Result,
+    },
+};
+
+#[derive(Clone)]
+pub enum JoinType {
+    Inner,
+    Left,
+}
+
+#[derive(Clone)]
+pub enum PrevNode {
+    Select(SelectNode),
+    Join(Box<JoinNode>),
+    JoinConstraint(Box<JoinConstraintNode>),
+}
+
+impl Prebuild for PrevNode {
+    fn prebuild(self) -> Result<NodeData> {
+        match self {
+            Self::Select(node) => node.prebuild(),
+            Self::Join(node) => node.prebuild(),
+            Self::JoinConstraint(node) => node.prebuild(),
+        }
+    }
+}
+
+impl From<SelectNode> for PrevNode {
+    fn from(node: SelectNode) -> Self {
+        PrevNode::Select(node)
+    }
+}
+
+impl From<JoinNode> for PrevNode {
+    fn from(node: JoinNode) -> Self {
+        PrevNode::Join(Box::new(node))
+    }
+}
+
+impl From<JoinConstraintNode> for PrevNode {
+    fn from(node: JoinConstraintNode) -> Self {
+        PrevNode::JoinConstraint(Box::new(node))
+    }
+}
+
+#[derive(Clone)]
+pub struct JoinNode {
+    prev_node: PrevNode,
+    relation: TableFactor,
+    join_operator: JoinOperator,
+    join_operator_type: JoinType,
```

Comment:
This `JoinNode` does not have JoinOperator with `ON` condition.

Only `JoinOperator::Inner(JoinConstraint::None) | JoinOperator::LeftOuter(JoinConstraint::None)` are being used.
Then, we don't need `join_operator: JoinOperator` field in `JoinNode` because that field can be simply generated from `join_operator_type: JoinType`.